src/loader/loader.c

```diff
@@ -335,10 +335,16 @@ post_analyze_hook(ParseState *pstate, Query *query)
 		switch (nodeTag(query->utilityStmt))
 		{
 			case T_DropdbStmt:
+
+				/*
+				 * Don't check if extension exists here because even though
+				 * the current database might not have TimescaleDB installed
+				 * the database we are dropping might.
+				 */
 				stop_workers_on_db_drop((DropdbStmt *) query->utilityStmt);
 				break;
 			case T_DropStmt:
-				if (drop_statement_drops_extension((DropStmt *) query->utilityStmt))
+				if (extension_exists() && drop_statement_drops_extension((DropStmt *) query->utilityStmt))
```

Comment:
I think I prefer to have these calls as part of the drop_xxxx functions, but I'll let you decide.

Reply:
I agree